src/uwtools/tests/config/test_validator.py

```diff
@@ -135,13 +136,123 @@ def test_validate_workflow_tags_pass(vals):
+            )
+            is False
+        )
+        assert logged(caplog, "'command' is a required property")
+
+
+def test_validate_yaml_rocoto_invalid_no_task(rocoto_assets, caplog):
+    schema_file, config_dict = rocoto_assets
+    del config_dict["workflow"]["tasks"]["metatask"]["task"]
+    with patch.object(validator, "YAMLConfig") as YAMLConfig:
+        yaml_config = YAMLConfig()
+        yaml_config.data = config_dict
+        assert (
+            validator.validate_yaml(
+                schema_file=schema_file, config_file="/not/used", check_path=False
+            )
+            is False
+        )
+        assert logged(caplog, "{'var': {'member': 'foo bar baz'}} does not have enough properties")
+
+
+def test_validate_yaml_rocoto_invalid_no_var(rocoto_assets, caplog):
+    schema_file, config_dict = rocoto_assets
+    del config_dict["workflow"]["tasks"]["metatask"]["var"]
+    with patch.object(validator, "YAMLConfig") as YAMLConfig:
+        yaml_config = YAMLConfig()
+        yaml_config.data = config_dict
+        assert (
+            validator.validate_yaml(
+                schema_file=schema_file, config_file="/not/used", check_path=False
+            )
+            is False
+        )
+        assert logged(caplog, "'var' is a required property")
+
+
+def test_validate_yaml_rocoto_invalid_type(rocoto_assets, caplog):
+    schema_file, config_dict = rocoto_assets
+    config_dict["workflow"]["tasks"]["metatask"]["task"]["cores"] = "string"
+    with patch.object(validator, "YAMLConfig") as YAMLConfig:
+        yaml_config = YAMLConfig()
+        yaml_config.data = config_dict
+        assert (
+            validator.validate_yaml(
+                schema_file=schema_file, config_file="/not/used", check_path=False
+            )
+            is False
+        )
+        assert logged(caplog, "'string' is not of type 'integer'")
+
+
+def test_validate_yaml_rocoto_invalid_walltime_pattern(rocoto_assets, caplog):
+    schema_file, config_dict = rocoto_assets
+    config_dict["workflow"]["tasks"]["metatask"]["task"]["walltime"] = "0:01:00"
+    with patch.object(validator, "YAMLConfig") as YAMLConfig:
+        yaml_config = YAMLConfig()
+        yaml_config.data = config_dict
+        assert (
+            validator.validate_yaml(
+                schema_file=schema_file, config_file="/not/used", check_path=False
+            )
+            is False
+        )
+        assert logged(caplog, "'0:01:00' does not match '^[0-9]{2}:[0-9]{2}:[0-9]{2}$'")
+
+
+def test_validate_yaml_rocoto_valid(rocoto_assets):
+    schema_file, config_dict = rocoto_assets
+    with patch.object(validator, "YAMLConfig") as YAMLConfig:
+        yaml_config = YAMLConfig()
+        yaml_config.data = config_dict
+        assert (
+            validator.validate_yaml(
+                schema_file=schema_file, config_file="/not/used", check_path=False
+            )
+            is True
+        )
```

Comment:
Nice work expanding these tests!

We could DRY this out a little more by having the fixture do, say,
```
return {"schema_file": schema_file, "config_file": "/not/used", "check_path": False}, { ... same big config dict ...}
```
unpacking `rocoto_assets` as
```
kwargs, config_dict = rocoto_assets
```
in each test function, and then doing the `validate_yaml()` calls like
```
assert validator.validate_yaml(**kwargs)) is ...
```
Since we (I think) make the `validate_yaml()` call exactly the same way in each test function, we could make the keyword arguments we pass to it part of the fixture and DRY that out. That might collapse about 6 lines in each test function down to 1, as it'll be a lot shorter.

You could also consider doing `assert` and `assert not` as shorthand, though I know that's not what I did when we were pairing...

Reply:
In the fixture, if you want to be more explicit, you could also do
```
kwargs = {"schema_file": schema_file, "config_file": "/not/used", "check_path": False}
config_dict = { ... same big config dict ... }
return kwargs, config_dict
```
That also introduces some consistent names between the variables that are packed into the fixture's return value and the variables they are unpacked into in the test functions. Might be nice.

And maybe it should just be called `config` and not `config_dict` -- we usually try to avoid encoding type info in variable names. (I know, my fault from our pairing session...)

Reply:
I'm trying to fully understand what you mean by:

"Since we (I think) make the validate_yaml() call exactly the same way in each test function, we could make the keyword arguments we pass to it part of the fixture and DRY that out. That might collapse about 6 lines in each test function down to 1, as it'll be a lot shorter."

I think you mean something like this:
```
kwargs = {"schema_file": schema_file, "config_file": "/not/used", "check_path": False}
config_dict = { ... same big config dict ... }
validate = 
    yaml_config = YAMLConfig()
        yaml_config.data = config
        assert not (
            validator.validate_yaml(
                **kwargs
            )
        )
return kwargs, config, validate
```

And then within the test case:
```
with patch.object(validator, "YAMLConfig") as YAMLConfig:
        validate
```

Reply:
What you did in https://github.com/ufs-community/workflow-tools/pull/293/files#diff-362fb561e035f3c6d293bd5b896a5fee3e568a9ca70dde56a1a6027224cb13ecR151-R157 is just what I had in mind. I think that's about as concise as we can get.